src/libethash/ethash.h

```diff
@@ -81,55 +73,45 @@ struct ethash_full;
 typedef struct ethash_return_value {
 	ethash_h256_t result;
 	ethash_h256_t mix_hash;
-} ethash_return_value;
+} ethash_return_value_t;
 
-uint64_t ethash_get_datasize(uint32_t const block_number);
-uint64_t ethash_get_cachesize(uint32_t const block_number);
+uint64_t ethash_get_datasize(uint64_t const block_number);
+uint64_t ethash_get_cachesize(uint64_t const block_number);
 
-// initialize the parameters
-static inline void ethash_params_init(ethash_params* params, uint32_t const block_number)
-{
-	params->full_size = ethash_get_datasize(block_number);
-	params->cache_size = ethash_get_cachesize(block_number);
-}
-
-// LTODO: for consistency's sake maybe use ethash_cache_t?
 typedef struct ethash_cache {
 	void* mem;
-} ethash_cache;
+	uint64_t cache_size;
+} ethash_cache_t;
 
 /**
  * Allocate and initialize a new ethash_cache object
  *
- * @param params    The parameters to initialize it with. We are interested in
- *                  the cache_size from here
- * @param seed      Block seedhash to be used during the computation of the
- *                  cache nodes
- * @return          Newly allocated ethash_cache on success or NULL in case of
- *                  ERRNOMEM or invalid parameters used for @ref ethash_compute_cache_nodes()
+ * @param cache_size    The size of the cache in bytes
+ * @param seed          Block seedhash to be used during the computation of the
+ *                      cache nodes
+ * @return              Newly allocated ethash_cache on success or NULL in case of
+ *                      ERRNOMEM or invalid parameters used for @ref ethash_compute_cache_nodes()
  */
-ethash_cache* ethash_cache_new(ethash_params const* params, ethash_h256_t const* seed);
+ethash_cache_t* ethash_cache_new(uint64_t cache_size, ethash_h256_t const* seed);
 /**
  * Frees a previously allocated ethash_cache
  * @param c            The object to free
  */
-void ethash_cache_delete(ethash_cache* c);
+void ethash_cache_delete(ethash_cache_t* c);
 
 /**
  * Allocate and initialize a new ethash_light handler
  *
- * @param params    The parameters to initialize it with. We are interested in
- *                  the cache_size from here
- * @param seed      Block seedhash to be used during the computation of the
- *                  cache nodes
- * @return          Newly allocated ethash_light handler or NULL in case of
- *                  ERRNOMEM or invalid parameters used for @ref ethash_compute_cache_nodes()
+ * @param cache_size    The size of the cache in bytes
+ * @param seed          Block seedhash to be used during the computation of the
+ *                      cache nodes
+ * @return              Newly allocated ethash_light handler or NULL in case of
+ *                      ERRNOMEM or invalid parameters used for @ref ethash_compute_cache_nodes()
  */
-ethash_light_t ethash_light_new(ethash_params const* params, ethash_h256_t const* seed);
+ethash_light_t ethash_light_new(uint64_t cache_size, ethash_h256_t const* seed);
```

Comment:
there should be no need to pass `cache_size`: everything can be calculated from `seed`.